docs/source/notes/autograd.rst

```diff
@@ -58,6 +58,100 @@ will also require them.
+
+Computations are performed in grad mode by default. In grad mode computations
+on tensors that require grad are tracked by autograd so that gradients can be
+computed with respect to the inputs when running ``output.backward()``. Note: this
+is the only mode where requires_grad is relevant. ``requires_grad`` is ignored in
+all other modes.
+
+Setting requires_grad
+^^^^^^^^^^^^^^^^^^^^^
+
+As described `above <excluding-requires_grad_>`_,
+use ``.requires_grad`` for more fine-grained control over which parameters should
+be updated on the backward pass. Parameters with ``requires_grad=True`` will have
+gradients accumulated into their ``.grad`` fields on ``.backward()``.
+
+Whether a tensor ``requires_grad`` also determines whether the computations are
+tracked by autograd in the first place. An operation is only tracked by
+autograd if at least one of its input tensors require grad.
+
+For example, if you need to freeze the bottom part of your model while
+fine-tuning, you can apply ``requires_grad=False`` on layers that you’d like to
+avoid.
+
+Note: you can only set the requires_grad of leaf nodes (i.e., parameters and
+other tensors that don’t have ``grad_fn``). Non-leaf tensor have ``requires_grad=True``
+by definition.
+
+No-grad Mode
+^^^^^^^^^^^^
+
+Computations in no-grad mode behave as if none of the inputs require grad.
+Enable no-grad mode when you need to perform operations that should not be
+tracked by autograd, but you’d still like to use the outputs of these
+computations in grad mode later. If you know that the outputs won’t be involved
+in computations in grad mode later, consider using inference mode.
+
+For example, you would use no-grad if you are writing an optimizer, you’d like to update parameters
+in-place in order to perform a training step without the update being tracked
+by autograd. You also intend to use the updated parameters for computations in
+grad mode in the next forward pass.
+
+Inference Mode
+^^^^^^^^^^^^^^
+
+Inference mode (PyTorch >= 1.9) provides improved performance during the model
+evaluation phase in exchange for somewhat narrower application. Inference mode
+is similar to no-grad mode in that operations performed in inference mode will
+not be tracked by autograd. But for improved performance, computations in
+inference mode skips over more autograd-specific code.
+
+Enable inference mode when you are performing computations that don’t need
+tracked for the gradient computation, AND you don’t plan on using the tensors
+created in inference mode in any computation that is tracked by autograd later.
+
+It is recommended that you try out inference mode. If it works out of the box
+for your use case it’s a free performance win. If you run into errors after
+enabling inference mode, you may need to be more careful about how you use
+tensors created in inference mode outside of inference mode. It is also
+possible that inference mode does not work for your use case. If so, consider
```

Comment:
It is now: ```
If you run into errors after
enabling inference mode, check that you are not using tensors created in
inference mode outside inference mode. It may also be possible that inference
mode does not apply to your use case, in which case you can always switch
back to no-grad mode.
```
The main issue still is that we want a single sentence to encapsulate what restriction inference mode brings.
I feel like what I have right now: "check that you are not using tensors created in inference mode outside inference mode" is  still a little misleading

Now I'm thinking it should be "check that you are not using tensors created in inference mode in computations tracked by autograd after exiting inference mode" but that be an exact copy of what I had above.

Reply:
I think that sentence from before is ok. The full thing would be: "If you run into errors after
enabling inference mode, check that you are not using tensors created in inference mode in computations tracked by autograd after exiting inference mode. If you cannot avoid such use in your case, you can always switch
back to no-grad mode."